spark-jobs/src/main/scala/za/co/absa/enceladus/common/version/SparkVersionGuard.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common.version
+
+import za.co.absa.commons.version.Version
+import za.co.absa.commons.version.impl.SemVer20Impl.SemanticVersion
+import za.co.absa.enceladus.common.SparkCompatibility
+
+object SparkVersionGuard {
+
+  /**
+   * Populates the version guard with the defaults from [[za.co.absa.enceladus.common.SparkCompatibility]]
+   */
+  def fromDefaultSparkCompatibilitySettings: SparkVersionGuard =
+    SparkVersionGuard(SparkCompatibility.minSparkVersionIncluded, SparkCompatibility.maxSparkVersionExcluded)
+}
+
+/**
+ * Setup Spark version guard with allowed min & max sem-ver version. Note that:
+ *  - min version is included to be allowed and may be non-final
+ *  - max version is excluded from allowed (supremum) and may be non-final
```

Comment:
Sumpremum?

Reply:
Well, the issue/feature description (further explained by @yruslan) was that we want to set bounds .e.g. 2.4.2-3.0.0, where the lower is in fact minimum (allowed), while the "max" is what is already excluded. So everything up to a "maximum" = [supremum](https://cs.wikipedia.org/wiki/Supremum).

Reply:
Ok, I didn't know the word - not in Czech nor in English 😊